agimus_controller_ros/agimus_controller_ros/simple_trajectory_publisher.py

```diff
@@ -37,59 +40,33 @@ def get_reduced_configuration(q: list[np.float64], joint_idxs: list[np.int64]):
     return reduced_q
 
 
-class QuinticTrajectory:
-    """Computes a quintic polynomial trajectory with desired amplitude and duration."""
-
-    def __init__(self, scale_duration: np.float64, amp: np.float64):
-        """Initialize polynomial attributes."""
-        self.amp = amp
-        self.scale_duration = scale_duration
-
-    def get_value_at_t(
-        self, t: np.float64
-    ) -> Tuple[np.float64, np.float64, np.float64]:
-        """Return polynomial value and his derivatives at time t."""
-        if t <= 0:
-            return 0.0, 0.0, 0.0
-        elif t >= self.scale_duration:
-            return self.amp, 0.0, 0.0
-
-        # Normalize time
-        s = t / self.scale_duration
-
-        polynomial = self.amp * (10 * s**3 - 15 * s**4 + 6 * s**5)
-        d_polynomial = (
-            self.amp * (30 * s**2 - 60 * s**3 + 30 * s**4) / self.scale_duration
-        )
-        dd_polynomial = (
-            self.amp * (60 * s - 180 * s**2 + 120 * s**3) / (self.scale_duration**2)
-        )
-        return polynomial, d_polynomial, dd_polynomial
-
-
 class SimpleTrajectoryPublisher(Node):
     """This is a simple trajectory publisher for a Panda robot."""
 
     def __init__(self):
         super().__init__("simple_trajectory_publisher")
 
-        self.pin_model = None
-        self.pin_data = None
-        self.ee_frame_id = None
         self.param_listener = trajectory_weights_params.ParamListener(self)
         self.params = self.param_listener.get_params()
         self.ee_frame_name = self.params.ee_frame_name
         self.robot_description_msg = None
 
         self.q0 = None
-        self.q = None
-        self.dq = None
-        self.ddq = None
+        self.current_q = None
         self.t = 0.0
         self.dt = 0.01
         self.croco_nq = 7
-        self.quint_traj = QuinticTrajectory(scale_duration=0.8, amp=0.2)
-        self.w = 0.5 * np.pi
+        self.trajectory = SinusWaveConfigurationSpace(
+            w=0.5 * np.pi,
+            scale_duration=0.8,
+            amp=0.2,
```

Comment:
If we go your way, can you make those parameter accessible either as cli args or as ROS params?